### PR TITLE
[MODULAR] Fixes the gear harness not being able to support attachments

### LINKED ISSUE #20214: Pride Pins don't spawn from loadout
Reporting client version: 514.1589

## Round ID:
7992

## Testmerges:
- [Replace Chaos Event System with ICES - Intensity Credit Event System](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19505)
- [Antag Revamp*: The Nuclear-ish Option](https://github.com/Skyrat-SS13/Skyrat-tg/pull/18110)
- [[TM ONLY] Disables Sideplane, Makes the Game Playable](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19308)
- [[TM-ONLY] Prevents the psyker pirate variant from rolling](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19820)
- [[EVIL CODE] [LATE MEOWY CHRISTMAS] Replaces the poorly fitting ashwalkers on icebox with significantly cooler (I can't think of an eyecatching title for demis alright?)](https://github.com/Skyrat-SS13/Skyrat-tg/pull/18364)

## Reproduction:

Have a pride pin in loadout > no pride pin at roundstart or on spawn. dire bug. pls fix.... 

*EDIT* Actual bug/stealth nerf is that gear harnesses no longer allow you attach accessories, such as pride pins. Clearly a mistake bcuz gear harnesses are mad gay.

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Fixes https://github.com/Skyrat-SS13/Skyrat-tg/issues/20214

What it says on the tin. The gear harness does not cover any body parts, so accessories will not work. To this end I added a new var, `accessory_slot_override` which can be used to define an accessory slot that is different from `body_parts_covered`.

## How This Contributes To The Skyrat Roleplay Experience

Bugfix

## Proof of Testing

<details>
<summary>It works</summary>
  
image

image

</details>

## Changelog

:cl:
fix: accessories may now be attached to gear harnesses
code: adds support for defining an explicit accessory slot for any given clothing, using the `accessory_slot_override` var
/:cl:

### CODE REVIEW

**Comment by kittysmooch** [comment]
gay rights 🙇

File: modular_skyrat/master_files/code/modules/clothing/under/accessories.dm [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/obj/item/clothing/accessory/can_attach_accessory(obj/item/clothing/clothing_item, mob/user)
+	if(!attachment_slot || (clothing_item && clothing_item.attachment_slot_override & attachment_slot))
```
Comment (tf-4) [inline modular_skyrat/master_files/code/modules/clothing/under/accessories.dm]:
```suggestion
	if(!attachment_slot || (clothing item?.attachment_slot_override & attachment_slot))
```
What situations do you have null clothing anyway? Does it still run if you hit a naked mob?

Comment (vinylspiders) [reply modular_skyrat/master_files/code/modules/clothing/under/accessories.dm]: Not sure honestly--was just quickly going by the original proc.

https://github.com/Skyrat-SS13/Skyrat-tg/blob/3a5711fb7fb5eee40acdaa3df3c4ef427be580a6/code/modules/clothing/under/accessories.dm#L17-L18

Upon taking a closer look it looks like the original coder was just being overzealous with the safety checking. Which I can get behind honestly, it's good futureproofing.

 -shrug-

File: modular_skyrat/master_files/code/modules/clothing/under/accessories.dm [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/obj/item/clothing/accessory/can_attach_accessory(obj/item/clothing/clothing_item, mob/user)
+	if(!attachment_slot || (clothing item?.attachment_slot_override & attachment_slot))
```
Comment (tf-4) [inline modular_skyrat/master_files/code/modules/clothing/under/accessories.dm]:
```suggestion
	if(!attachment_slot || (clothing_item?.attachment_slot_override & attachment_slot))
```
I am forced to admit that my previous suggestion was done on my phone

**Review by tf-4 — APPROVED** [review]

**Pull request merged**